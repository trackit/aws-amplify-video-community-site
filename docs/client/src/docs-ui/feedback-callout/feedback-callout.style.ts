import {css} from "emotion";

export const hostStyle = css`
  display: none;
  position: relative;
  margin-top: 0.5rem;

  button {
    position: absolute;
    top: 0.5rem;
    right: 0.5rem;
    appearance: none;
    cursor: pointer;
    background-color: transparent;

    img {
      min-width: 0.75rem;
      min-height: 0.75rem;
    }
  }

  > amplify-external-link {
    display: flex;
    border-radius: 0.25rem;
    background-color: var(--color-ink-hv);

    a {
      width: 100%;
      display: flex;
      flex-direction: column;
      padding: 0.75rem 1rem;
      color: var(--color-white);
    }
  }
`;

export const displayStyle = css`
  display: block;
`;

export const arrowStyle = css`
  display: inline-block;
  height: 0.125rem;
  width: 0.125rem;
  border: solid white;
  border-width: 0 0.125rem 0.125rem 0;
  padding: 0.1875rem;
  transition: all 0.25s cubic-bezier(0.165, 0.84, 0.44, 1);
`;

export const arrowUpStyle = css`
  margin-top: 0.125rem;
  transform: rotate(-135deg);
`;

export const exStyle = css`
  margin: 0.125rem 0.125rem 0 0;
`;

export const calloutTextStyle = css`
  > i {
    margin-right: 0.875rem;
  }
  > .text {
    font-size: 0.875rem;
    font-weight: bold;
  }
`;
