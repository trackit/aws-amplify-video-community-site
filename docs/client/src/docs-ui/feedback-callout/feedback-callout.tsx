import {Component, h, Host, State} from "@stencil/core";
import {
  hostStyle,
  displayStyle,
  arrowStyle,
  arrowUpStyle,
  exStyle,
  calloutTextStyle,
} from "./feedback-callout.style";
import {createIssueLink} from "../../utils/issue-link";

const IS_DISMISSED_LOCAL_STORAGE_KEY =
  "amplify-docs::feedback-request-dismissed";

@Component({tag: "docs-feedback-callout", shadow: false})
export class DocsFeedbackCallout {
  @State() dismissed = true;

  componentWillLoad() {
    this.dismissed = !!localStorage.getItem(IS_DISMISSED_LOCAL_STORAGE_KEY);
  }

  dismiss = () => {
    this.dismissed = true;
    localStorage.setItem(IS_DISMISSED_LOCAL_STORAGE_KEY, "true");
  };

  render() {
    return (
      <Host class={{[hostStyle]: true, [displayStyle]: !this.dismissed}}>
        <amplify-external-link href={createIssueLink()}>
          <span class={calloutTextStyle}>
            <i
              class={{
                [arrowStyle]: true,
                [arrowUpStyle]: true,
              }}
            ></i>
            <span class="text">{`We'd love your feedback`}</span>
          </span>
        </amplify-external-link>
        <button class={exStyle} onClick={this.dismiss}>
          <img src="/assets/close.svg" />
        </button>
      </Host>
    );
  }
}
