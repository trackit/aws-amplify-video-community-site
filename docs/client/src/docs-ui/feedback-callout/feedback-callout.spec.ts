import {DocsFeedbackCallout} from "./feedback-callout";
import {newSpecPage} from "@stencil/core/testing";

describe("docs-feedback-callout", () => {
  describe("Render logic", () => {
    it("should render", async () => {
      expect(
        (
          await newSpecPage({
            components: [DocsFeedbackCallout],
            html: `<docs-feedback-callout />`,
          })
        ).root,
      ).toMatchSnapshot();
    });
  });
});
