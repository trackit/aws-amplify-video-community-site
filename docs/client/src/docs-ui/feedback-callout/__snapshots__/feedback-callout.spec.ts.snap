// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`docs-feedback-callout Render logic should render 1`] = `
<docs-feedback-callout class="css-1i82ydo css-doub5p">
  <amplify-external-link href="https://github.com/aws-amplify/docs/issues/new?title=[Feedback]FEEDBACK_TITLE_HERE&amp;labels=&amp;body=**Page**%3A%20%5B%60%2F%60%5D(http%3A%2F%2Ftesting.stenciljs.com%2F)%0A%0A**Feedback**%3A%0A%0A%3C!--%20your%20feedback%20here%20--%3E">
    <span class="css-9e4xma">
      <i class="css-1byxedo css-354oya"></i>
      <span class="text">
        We'd love your feedback
      </span>
    </span>
  </amplify-external-link>
  <button class="css-nt7bpa">
    <img src="/assets/close.svg">
  </button>
</docs-feedback-callout>
`;
